File: test-client/src/models/viewModels/RoomVM.ts
import { IRoom, RoomType } from "models/interfaces/IRoom"

export interface RoomVM extends IRoom {
    _id: string,
    type: RoomType,
    name?: string,
    displayPicture?: string,

    messages: string[]
    users: string[]
}


/* 
    RoomVM does not have attributes:
        - pin
*/

File: websocket/model/view_model/RoomVM.py
from dataclasses import dataclass, field
from typing import Optional, List

@dataclass
class RoomVM():
    _id: str
    type: str
    name: Optional[str] = None
    displayPicture: Optional[str] = None

    messages: List[str] = field(default_factory=list)
    users: List[str] = field(default_factory=list)


# RoomVM does not have:
#   - pin